### PR TITLE
Added ThrowsCount rule

### PR DESCRIPTION
**Pull request by schalkms**

### CODE REVIEW

File: detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/ThrowsCount.kt [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+	override val issue = Issue(javaClass.simpleName, Severity.Style,
+			"Restrict the number of throw statements in methods.",
+			Debt.FIVE_MINS)
+
+	private val max = valueOrDefault(MAX, 2)
+
+	override fun visitNamedFunction(function: KtNamedFunction) {
+		super.visitNamedFunction(function)
+		if (!function.hasModifier(KtTokens.OVERRIDE_KEYWORD)) {
```
Comment (vanniktech) [inline detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/ThrowsCount.kt]: let's also add a test for this one?

File: detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/ThrowsCount.kt [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+			val count = function.collectByType<KtThrowExpression>().count()
+			if (count > max) {
```
Comment (vanniktech) [inline detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/ThrowsCount.kt]: is this the right behavior? I think there was a discussion whether 2:2 is fine or not

Comment (arturbosch) [reply detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/ThrowsCount.kt]: It is the right behavior until the issue is solved. The `actual > threshold` I used everywhere in detekt ^^

Comment (vanniktech) [reply detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/ThrowsCount.kt]: Should we extract this into a method so if we decide to change it we'll only have to change it in one place?

Comment (schalkms) [reply detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/ThrowsCount.kt]: I think it makes sense of course. From a user perspective I wouldn't expect that the threshold changes. A user would have to increment all threshold values in the detekt config.

Comment (arturbosch) [reply detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/style/ThrowsCount.kt]: I would just say leave it as it is until we get more complaining or something ... :D

**Pull request merged**